Makefile

```diff
@@ -19,8 +19,8 @@ endif
 coverage-report: ## coverage-report: Run the unit tests with code coverage enabled.
 	unset FT_NEXT_BACKEND_KEY && FT_GRAPHITE_KEY=foobar istanbul cover node_modules/.bin/_mocha --report=$(if $(CIRCLECI),lcovonly,lcov) 'test/**/*.test.js'
 
+## configure unit test to run app.test.js with --no-exit, to demonstrate the tests stop hanging by calling app.close()
 unit-test:
-	unset FT_NEXT_BACKEND_KEY && FT_GRAPHITE_KEY=foobar mocha test/**/*.test.js
-
+	unset FT_NEXT_BACKEND_KEY && export FT_GRAPHITE_KEY=foobar && mocha --no-exit test/app/clear-interval.test.js && mocha test/**/*.test.js --exclude test/app/clear-interval.test.js
```

Comment:
Great question! TLDR: I've included the fix in the newest commit.

To be honest, I was so tempted to just not spend any more time on fixing tests and therefore allow the other tests to use the --exit option, because after adding the app.close() they were still hanging, and I've spent too long investigating already. It was hard to pinpoint which test was causing it to hang, because when running a test file individually eg. app.test.js
without any of my changes, some tests in it fail and I reckon it's because of some setup in before hooks between different test files are interleaved which means some tests depend on some tests but the dependencies aren't clear. Also wtfnode didn't help me pinpoint which process caused it to hang because it only showed me the mocha stack trace. 
Eventually I realised the hanging was coming from n-flags-client which the test fixture depended on, so clearing that interval caused the tests to not hang with the --no-exit option